File: python/problem002.py
result = 0
index = 0
true = 1


def fibonacci(n):
    if n == 1:
        return 1
    elif n == 2:
        return 2
    elif n > 2:
        return fibonacci(n - 2) + fibonacci(n - 1)


while True:
    index += 1
    if fibonacci(index) < 4000000:
        if fibonacci(index) % 2 == 0:
            result += fibonacci(index)
    else:
        break

print(result)

File: java/src/Problem002.java
/**
 * Created by 김 on 2016-02-05.
 */
public class Problem002 {

    int result(){
        int value = 0;
        int x = 1;
        int y = 2;
        while(y < 4000000){
            if(y % 2 == 0){
                value = value + y;
            }
            int temp = y;
            y = x + y;
            x = temp;
        }
        return value;
    }

    public static void main(String args[]){
        Problem002 problem = new Problem002();
        System.out.println(problem.result());

    }
}
